tests/test_pay.py

```diff
@@ -3261,6 +3261,77 @@ def test_mpp_presplit_routehint_conflict(node_factory, bitcoind):
     l1.rpc.pay(inv)
 
 
+def test_delpay_argument_invalid(node_factory, bitcoind):
+    """
+    This test includes all possible combination of input error inside the
+    delpay command.
+    """
+
+    l1, l2 = node_factory.get_nodes(2)
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay()
+
+    # invoice unpaid
+    inv = l1.rpc.invoice(10 ** 5, 'inv', 'inv')
+    payment_hash = inv["payment_hash"]
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash)
+
+    # payment unpaid with wrong status (pending status is a illegal input)
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'pending')
```

Comment:
Payment was no attempted at all, so it'd fail for that reason already.

Reply:
Basically this is testing our sanity check order, and not the underlying `pending` vs. `non-pending` cases.

Reply:
This comment brings me to reflect on the sens of this test!

I moved the pay command to the top of this sanity check, so, with this new situation the unit test has more sense to me because it does the sanity check and the command behavior checks